### ISSUE TITLE
Question: multiple tables

### ISSUE THREAD

**Issue by footh** [awslabs/aws-lambda-redshift-loader#99]
What's the best way to do multiple tables? Say I have two files cats.csv and dogs.csv that need to go into the tables `cats` and `dogs` respectively. Both of these files are located in the same bucket and folder (ex. my_bucket/animals/).  

Seems to me the only way is to have each file in a different folder and create a config for each one?

**Comment by IanMeyers** [comment]
You'd want to have 2 configurations for the top level prefix, and then a separate filter expression for `cats` and `dogs`, which can then be routed to separate clusters/tables.

**Comment by footh** [comment]
Is it possible to have 2 configs with the same top level prefix? The prefix has a uniqueness constraint it would appear.

**Comment by IanMeyers** [comment]
Sorry - you are correct - the filter expression I cited would have to be part of the input prefix you specify on the config.

Ian

**Comment by footh** [comment]
I don't think that solution works either since the cats.csv and dogs.csv  files are in the same directory and the prefix used to resolve a config removes the file name. The only way I see is to have those files in different directories. Or a hack-ish solution is to have those files in a deeper directory (say, `mybucket/data/animals`) and have a config for `mybucket/data` and `mybucket/data/animals` with file regex filters to pick up the desired files for each config.

I did create a branch that adds an optional config entry that looks like this:

```
  "tableFilterMap": {
    "cats\\.csv": "cats",
    "dog\\.csv": "dogs"
  }
```

Which will run through the regex and pick the right table for the copy command (or default to the  `table` parameter if no regex matches or the parameter is empty). But I've discovered this doesn't work with batches since the copy command uses the last table found in the batch file list. It mostly works with a batch size of 1 but in our test runs, I've noticed there are still times when a batch has two files in it which of course results in an error.

Do you have any suggestions on a better way to do this? I'm happy to work on this feature since it's not an option for us to muck with the s3 directory structure to get this to work.

**Comment by IanMeyers** [comment]
We can try to address this directly in the loader still, but another option would be to route S3 event notifications to a Lambda function that moves the s3 files to a prefix structure that would separate files by table destination. You then create separate config a for these prefixes.  Would that be worth considering?

**Comment by footh** [comment]
I don't think we want to add more complexity to the process by adding another lambda function either. Turns out we have more leeway than I thought to arrange the files mainly because we need to get this initial batch done. It's not ideal as it requires duplicating a lot of directory structure but it will work for now.

Going forward that I wouldn't mind investigating a way to do this within the loader itself.
